### PR TITLE
Move TableDefinitions to AgentState (and some other things)

### PR DESCRIPTION
**Pull request by glarwood**

### CODE REVIEW

File: src/main/java/com/fivetran/agent/mysql/output/BucketOutput.java [hunk 1]
```diff
@@ -146,7 +123,7 @@ private void flushToBucket() throws IOException {
     }
 
     private void refreshFileChannel() {
-        path = Paths.get(DATA_FILE_PREFIX + Instant.now().getEpochSecond() + DATA_FILE_EXTENSION);
+        path = Paths.get(DATA_FILE_PREFIX + Instant.now().toEpochMilli() + DATA_FILE_EXTENSION);
```
Comment (glarwood) [inline src/main/java/com/fivetran/agent/mysql/output/BucketOutput.java:126]: With this, we shouldn't have to sleep all those test threads in MysqlUpdaterV4/MysqlAgentUpdater specs

File: src/test/java/com/fivetran/agent/mysql/BatchUpdaterTest.java [hunk 2]
```diff
@@ -24,7 +25,8 @@
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.junit.Assert.*;
 
-public class BatchUpdaterSpec {
+public class BatchUpdaterTest {
```
Comment (glarwood) [inline src/test/java/com/fivetran/agent/mysql/BatchUpdaterTest.java:28]: Maven won't run junit tests unless the enclosing class ends with 'Test'

File: src/test/java/com/fivetran/agent/mysql/BatchUpdaterTest.java [hunk 3]
```diff
@@ -188,12 +192,10 @@ public void binlogSync_updateTableDefinition() throws Exception {
         TableState tableState = new TableState();
 
         tableState.finishedImport = true;
-        state.tables.put(tableRef, tableState);
+        state.tableStates.put(tableRef, tableState);
+        state.tableDefinitions.put(tableRef, initialTableDef);
```
Comment (glarwood) [inline src/test/java/com/fivetran/agent/mysql/BatchUpdaterTest.java:196]: in `updateStateTableInfo()`we can always expect the number of TableStates to equal the number of TableDefinitions

**Review by IanWoods1993 — APPROVED** [review]
Great, it seems like this simplifies things a lot.

**Pull request merged**